server/routes/nl/helpers.py

```diff
@@ -104,6 +106,9 @@ def parse_query_and_detect(request: Dict, app: str, debug_logs: Dict):
   else:
     llm_api_type = LlmApiType(llm_api_type)
 
+  if i18n and i18n.lower() == 'true':
+    original_query = detect_lang_and_translate(original_query)
```

Comment:
Let's move `counters = ctr.Counters()` up, pass it into detect_lang_and_translate()

And in there, update `counters.info()` when translation occurs.  And also time it with `counters.timeit()`.

Reply:
Done. Didn't know about this - counters are great!

https://screenshot.googleplex.com/A4B4bCWjVfazjor